apache/superset
# fix(generic-chart-axes): apply time filter on all target column types

### SUMMARY
When the generic chart axes feature flag is enabled, temporal filters can only be applied on numeric target columns, causing problems for databases such as Druid which stores temporal values in `LONG` format. To fix this, we should short circuit the `handle_single_value` inner function in the `filter_values_handler` of the `BaseDatasource` class and return the filter value as-is, as that should always be passed to the `get_since_until_from_time_range` in it's original format [here](https://github.com/apache/superset/blob/22fab5e58ce574e962518067d982e3036449e580/superset/connectors/sqla/models.py#L1513-L1517).

As a follow-up, we should clean up these helper functions and add proper unit tests to them (adding unit tests in the current state of this code is fairly difficult).

### AFTER
After this fix, the temporal adhoc filter applies cleanly to a non-temporal column, similarly to how id does when disabling the "GENERIC_CHART_AXES" feature flag:
<img width="882" alt="image" src="https://user-images.githubusercontent.com/33317356/204217555-2c470553-85a1-4fcb-a80b-947e126c17d8.png">

### BEFORE
Currently the chart is unable to render, returning a "Must specify a value for filters with comparison operators" error:
<img width="877" alt="image" src="https://user-images.githubusercontent.com/33317356/204217342-879e94f9-e52b-4e67-b56a-1cf9b29a61c9.png">


### TESTING INSTRUCTIONS
1. Gnable "GENERIC_CHART_AXES` feature flag.
2. Go to "Video Game Sales" dashboard and edit the "Most Dominant Platforms" chart.
3. See an error. Set the time filter to "100 years ago" to see that the query generates a relevant filter after the fix has been applied.

### ADDITIONAL INFORMATION
